algoliasearch/client_test.go

```diff
@@ -430,3 +432,75 @@ func TestDnsTimeout(t *testing.T) {
 		t.Fatalf("TestDnsTimeout: Spent %d seconds instead of <5s to perform the 10 retries", int(delta.Seconds()))
 	}
 }
+
+func TestMultiClusterManagement(t *testing.T) {
+	client := initMCMClient(t)
+	var uniqueUserID = fmt.Sprintf("go-MCM-tests-%d", rand.Int())
+
+	t.Log("TestMultiClusterManagement: Listing clusters")
+	clusters, err := client.ListClusters()
+	if err != nil {
+		t.Fatalf("TestMultiClusterManagement: Cannot list Clusters: %s", err)
+	}
+	if len(clusters) <= 1 {
+		t.Log("The app as 1 or less clusters, which is odd for an MCM app")
+	}
+
+	clusterName := clusters[0].ClusterName
+
+	t.Logf("TestMultiClusterManagement: Assigning userID [%s] to cluster [%s]", uniqueUserID, clusterName)
+	_, err = client.AssignUserID(uniqueUserID, clusterName)
+	if err != nil {
+		t.Fatalf("TestMultiClusterManagement: Cannot assign a userIDs: %s", err)
+	}
+
+	t.Logf("TestMultiClusterManagement: Getting userID [%s]", uniqueUserID)
+	var userID UserID
+	for i := 0; i < 10; i++ {
+		userID, err = client.GetUserID(uniqueUserID)
+		if err == nil {
+			break
+		}
+		time.Sleep(time.Duration(i) * time.Second)
+	}
```

Comment:
I'm not sure what we are testing here? Why are we doing 10 times the same test + adding sleep times in between. Not only this makes our tests take more time to run, but it's also weird that no error is checked in the end.

Reply:
Unfortunately, MCM doesn't return taskID, so if the test is too fast, the userID is not yet available when we try to get it. I added this loop to try mulitple times and wait. In the end, if no userID could be found after the 10th try, it will file line 467.

Reply:
I don't think we want our users to do the same as you did here with a wait etc. What about making the `GetUserID` synchronous then, and encapsulate the retry internally, with a fix number of retries and finally return the error?